packages/node/src/indexer/fetch.service.ts

```diff
@@ -364,16 +368,18 @@ export class FetchService implements OnApplicationShutdown {
           const dictionary = await this.dictionaryService.getDictionary(
             startBlockHeight,
             queryEndBlock,
+            this.specVersionService.lastQueriedHeight,
             scaledBatchSize,
             this.dictionaryQueryEntries,
           );
-          //TODO
-          // const specVersionMap = dictionary.specVersions;
           if (
             dictionary &&
             this.dictionaryValidation(dictionary, startBlockHeight)
           ) {
-            const { batchBlocks } = dictionary;
+            const { batchBlocks, specVersions } = dictionary;
```

Comment:
You can get all chain upgrades in one query, since there won't be many. Then you don't need to query it in every batch